src/test/resources/com/puppycrawl/tools/checkstyle/checks/design/finalclass/InputFinalClass.java

```diff
@@ -127,3 +129,50 @@ private SomeClass() {}
         private SomeClass() {}
     }
 }
+
+class TestAnonymousInnerClasses { // ok
+    public static final TestAnonymousInnerClasses ONE = new TestAnonymousInnerClasses() {
```

Comment:
As mentioned above, show a case where `TestAnonymousInnerClasses` has a nested anonymous class of some class which is **NOT** `TestAnonymousInnerClasses`, which also has another nested anonymous class of `TestAnonymousInnerClasses`.

2 layer inception.

Reply:
done.